nautobot/core/tests/test_views.py

```diff
@@ -450,3 +456,31 @@ def test_get_object_with_constrained_permission(self):
         url = f"{reverse('db_file_storage.download_file')}?name={self.file_proxy_2.file.name}"
         response = self.client.get(url)
         self.assertHttpStatus(response, 404)
+
+
+class SettingsJSONSchemaViewTestCase(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        file_path = os.path.dirname(os.path.dirname(os.path.abspath(__file__))) + "/settings.json"
+        with open(file_path, "r") as jsonfile:
+            cls.json_data = json.load(jsonfile)
+
+    def test_settings_json_schema_valid(self):
+        try:
+            Draft7Validator.check_schema(self.json_data)
+        except SchemaError as e:
+            raise ValidationError({"data_schema": e.message})
+
+    def test_settings_json_schema_contains_valid_setting_variables(self):
+        TYPE_MAPPING = {
+            "string": [str],
+            "object": [dict],
+            "integer": [int],
+            "boolean": [bool],
+            "array": [list, tuple],
+            "#/definitions/callable": [types.FunctionType],
+        }
+        for key, value in self.json_data["properties"].items():
+            settings_value = getattr(settings, f"{key}", None)
+            if settings_value:
+                self.assertIn(type(settings_value), TYPE_MAPPING[value.get("type", value.get("$ref", None))])
```

Comment:
This is currently testing the variables that exist in the `settings.json` file. Some optional settings variables do not exist in `settings.py` and some variables exist in `settings.py` and not in `settings.json`.

Reply:
I'm surprised that we have variables in the schema that don't exist in some form in `settings.py` - I guess those might include the Constance-capable settings, but what others are missing?

Reply:
`CELERY_BROKER_TRANSPORT_OPTIONS`
`CONFIG_CONTEXT_DYNAMIC_GROUPS_ENALBED`
`SESSION_ENGINE`

Nautobot Version Control App Settings:

`DATABASE_ROUTERS`
`SESSION_CACHE_ALIAS`

Reply:
Most of the untested setting variables are included in `CONSTANCE_CONFIG` though.

Reply:
Might be good to have an explicit allow-list in the test for the above variables? So we can otherwise catch any case where the settings and the schema drift out of sync?

Reply:
Just to clarify, this allow-list is to catch setting variables that exist outside of those specified in the JSON schema right?